gslib/commands/signurl.py

```diff
@@ -403,6 +404,39 @@ class UrlSignCommand(Command):
       subcommand_help_text={},
   )
 
+  _COMMAND_MAP = GcloudStorageMap(
+      gcloud_command=[
+          'alpha', 'storage', 'sign-url',
+          '--format=csv[separator="\\t"](resource, http_verb, expiration, signed_url)'
+      ],
+      flag_map={
+          '-m': GcloudStorageFlag('--http-verb'),
+          '-d': GcloudStorageFlag('--duration'),
+          '-b': GcloudStorageFlag('--query-params=userProject'),
+          '-c': GcloudStorageFlag('--headers=content-type'),
+          '-r': GcloudStorageFlag('--region'),
+          '-p': GcloudStorageFlag('--private-key-password')
+      },
+  )
+
+  def get_gcloud_storage_args(self):
+    self._COMMAND_MAP.gcloud_command += ['--private-key-file=' + self.args[0]]
+    self.args = self.args[1:]
+
+    retention_arg_idx = 0
+    while retention_arg_idx < len(self.sub_opts):
+      if self.sub_opts[retention_arg_idx][0] == '-d':
+        break
+      retention_arg_idx += 1
+    if retention_arg_idx < len(self.sub_opts):
```

Comment:
I think it's a bit less error prone if you initially set the idx variable to `None`, set it on index match, and proceed with the conversion if not `None`.

Reply:
Done.